src/kibana-cf_authentication/server/routes.js

```diff
@@ -309,16 +310,20 @@ module.exports = (server, config, cache) => {
         handler: async (request, h) => {
           const options = {
             method: 'POST',
-            url: '/api/reporting/generate/csv',
+            url: '/api/reporting/generate/csv?jobParams=' + request.payload,
             artifacts: true
           };
           let cached
           try {
             cached = await cache.get(request.auth.credentials.session_id)
 
             if (cached.account.orgs.indexOf(config.get('authentication.cf_system_org')) === -1 && !(config.get('authentication.skip_authorization'))) {
+              // Requires Rison processing support to account for the jobParams payload
+              // See https://www.elastic.co/guide/en/kibana/7.x/reporting-integration.html
               let payload = JSON.parse(request.payload.toString() || '{}')
-              payload = filterCSVReportingQuery(payload, cached)
+              let decodedJobParams = rison.decode(payload.jobParams)
+              decodedJobParams = filterCSVReportingQuery(decodedJobParams, cached)
```

Comment:
Is there any reason this is just not assigned another variable name?

Reply:
Nope; fix incoming!